.github/workflows/main.yml

```
@@ -13,29 +13,39 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
   
-before_install:
-  
-  
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   test:
     # The type of runner that the job will run on
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-#        os: [ubuntu-18.04, macos-10.15, windows-2019]
+#        os: [ubuntu-18.04, macos-10.15]
         os: [ubuntu-18.04 ]
+#        compiler: [gfortran-8, gfortran-9]
+        compiler: [gfortran-9]
+    env:
+      FC: ${{ matrix.compiler }}
     steps:
     - name: Getting repository
       uses: actions/checkout@v2
     - name: Install dependencies
       run: |
         sudo apt-get install gfortran libblas-dev liblapack-dev m4
+        git clone -b v4.1.15 https://github.com/Goddard-Fortran-Ecosystem/pFUnit.git
+    - name: Build pFUnit
+      run: |
+          cd pFUnit
+          mkdir build
+          cd build
+          gfortran -v
+          cmake .. -DSKIP_MPI=yes -DSKIP_OPENMP=yes -DCMAKE_Fortran_COMPILER=${FC}
+          make install
```

```suggestion
          make install -j2
```
using 2 cores should hopefully reduce the CI compilation time a bit